### PR TITLE
feat(inputs.redis): support of redis 6.2 ERRORSTATS

### PR DESCRIPTION
**Pull request by kell0gg**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves [#13717](https://github.com/influxdata/telegraf/issues/13717)

### CODE REVIEW

File: plugins/inputs/redis/README.md [hunk 1]
```diff
@@ -164,6 +164,10 @@ and the elapsed time since the last rdb save (rdb\_last\_save\_time\_elapsed).
     - lag(int, number)
     - offset(int, number)
 
+- redis_errorstat_total
+  - tags:
+    - err(int, number)
+
```
Comment (srebhan) [inline plugins/inputs/redis/README.md:170]: Shouldn't this be
```suggestion
- redis_errorstat
  - tags:
    - err
  - fields:
    - total (int, number)
```
according to what you do in the code...

**Review by srebhan — COMMENTED** [review]
Thanks for your contribution @kell0gg! Some small comments...

File: plugins/inputs/redis/README.md (hunk 1 above)
Comment (kell0gg) [reply plugins/inputs/redis/README.md:170]: There was only one field, so I wasn't sure how to handle it.
I think you're right about separating the tag and the field.

File: plugins/inputs/redis/README.md [hunk 2]
```diff
@@ -217,3 +221,9 @@ redis_command:
 ```text
 redis_cmdstat,command=publish,host=host,port=6379,replication_role=master,server=localhost calls=68113i,usec=325146i,usec_per_call=4.77 1559227136000000000
 ```
+
+redis_error:
+
+```text
+redis_errorstat,err=MOVED,host=host,port=6379,replication_role=master,server=localhost total=4284 1691119309000000000
```
[the remark this answers is not in the document]
Comment (kell0gg) [reply plugins/inputs/redis/README.md:228]: port and server information is being collected [here](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/redis/redis.go#L287-L288),
replication_role is being collected [here](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/redis/redis.go#L455)


Here's the config file info and the metrics I got.

```
[[outputs.prometheus_client]]
  listen = ":9273"

[[inputs.redis]]
  servers = ["tcp://xxx.xxx.xxx.xxx:6379"]
```

```
# TYPE redis_errorstat_total untyped
redis_errorstat_total{err="CLUSTERDOWN",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 8
redis_errorstat_total{err="CROSSSLOT",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 3
redis_errorstat_total{err="ERR",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 175
redis_errorstat_total{err="LOADING",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 4284
redis_errorstat_total{err="MASTERDOWN",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 102
redis_errorstat_total{err="MOVED",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 3659
redis_errorstat_total{err="NOSCRIPT",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 4
redis_errorstat_total{err="WRONGPASS",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 2
redis_errorstat_total{err="WRONGTYPE",host="kell0g-1",port="6379",replication_role="master",server="xxx.xxx.xxx.xxx"} 31
```

[the remark this answers is not in the document]
Comment (srebhan) [reply plugins/inputs/redis/README.md:230]: Thanks for checking this! Much appreciated!

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for the nice PR @kell0gg!

**Review by powersj — APPROVED** [review]

**Pull request merged**